### DISCUSSION TITLE
CBReader 經文資料是否能收錄 cbeta-api 的目錄？

### DISCUSSION THREAD

**Discussion by MrMYHuang** [DILA-edu/cbeta-api#32] [General]
各位大德，您好：

我開發的一款 CBETA 電子佛典 app - [cbetar2](https://github.com/MrMYHuang/cbetar2) 支援 cbeta-api 連線存取目錄與經文檔，也支援離線讀取 [CBETA CBReader 經文資料檔](http://www.cbeta.org/download/cbreader.htm#data)。

經研究發現 cbeta-api 的 
[CBETA 部類目錄](https://cbdata.dila.edu.tw/v1.2/catalog_entry?q=CBETA)
與 CBReader 經文資料的 bulei_nav.xhtml 目錄結構幾乎一致，因此容易用程式建立相容的目錄書籤。

但[其它分類](https://cbdata.dila.edu.tw/v1.2/catalog_entry?q=root)，如冊分類：
https://cbdata.dila.edu.tw/v1.2/catalog_entry?q=vol
則沒有相對應的 CBReader 經文資料檔。因此我的 app 在從連線版切至離線版後，這些目錄書籤就無法使用。

至於經、卷的 APIs 部分與 CBReader 經文資料結構能夠對應，是 ok 的。
https://cbdata.dila.edu.tw/v1.2/static_pages/work
https://cbdata.dila.edu.tw/v1.2/static_pages/get_html

希望 CBReader 經文資料能提供 cbeta-api 的目錄資料。

**Comment by RayCHOU** [comment]
隆重介紹 CBReader 開發者 @heavenchou ^_^

**Comment by heavenchou** [comment]
CBReader 經文資料的 bulei_nav.xhtml 目錄中，還有其它的 *.xhtml
其中也有根據冊分類的 ，@MrMYHuang 可看看是否是您要的？

**Reply by MrMYHuang** [reply]
@heavenchou 您好，

我找過 Bookcase 的檔案，包括這個冊分類的目錄 advance_nav.xhtml，但它仍與 cbeta-api 的冊分類結構有差異。例如《中華大藏經》的 cbeta-api：
https://cbdata.dila.edu.tw/v1.2/catalog_entry?q=Vol-C
包含一個 C057 的項目，但搜尋 advance_nav.xhtml 找不到這個項目。

**Comment by heavenchou** [comment]
了解您的意思了。
因為 advance_nav.xhtml 是以經文為主，為了避免層次太多，所以在經之前就沒有冊的分類。而 C057 冊是 C056 冊的 1163 經的後半部，所以就沒有列出 C057冊了。
您可以看看 catalog.txt 這個檔案，因為我也需要查詢各冊有哪些經，就會用到這個檔。它裡面就有：

C , 事彙部類 , , 056 , 1163 , 15 , 一切經音義(第1卷-第15卷) , 唐 玄應撰
C , 事彙部類 , , 057 , 1163 , 10 , 一切經音義(第16卷-第25卷) , 唐 玄應撰

**Reply by MrMYHuang** [reply]
謝謝您，catalog.txt 的確有資訊由冊查詢經。但要將其重建出 cbeta-api 的冊分類目錄這就要研究看看了。

幸好 "CBETA 部分類"在 CBReader 經文資料與 cbeta-api 相容的。

**Comment by RayCHOU** [comment]
CBETA API 的目錄來自這裡：
https://github.com/DILA-edu/cbeta-metadata/tree/master/catalog

每季更新的時候，我會參考 CBReader 的目錄資料，
更新上述 GitHub 上的 metadata.

當時在 GitHub 上製作 CBETA Metadata,
就是希望有一份容易編輯維護、Machine Readable 的共用資料。

**Reply by MrMYHuang** [reply]
謝謝。
